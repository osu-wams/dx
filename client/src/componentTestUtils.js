import React from 'react';
import { render as rtlRender } from 'react-testing-library';
import { createHistory, createMemorySource, LocationProvider } from '@reach/router';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

function render(ui, ...rest) {
  return rtlRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, ...rest);
}

// this is a handy function that I would utilize for any component
// that relies on the router being in context
function renderWithRouter(
  ui,
  { route = '/', history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...rtlRender(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us to reference it in our tests
    history
  };
}

export * from 'react-testing-library';
// override the built-in render with our own
export { render, renderWithRouter };
