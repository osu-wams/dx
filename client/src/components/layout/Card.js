import React, { useState, useContext, useRef, useEffect } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { shadows, colors } from '../../theme';
import useMediaQuery from '../../util/useMediaQuery';

const CardContext = React.createContext();

/*===========================================
                    CARD
===========================================*/
const Card = ({ children, color, ...props }) => {
  const [collapsed, setCollapsed] = useState(true);

  return (
    <CardContext.Provider
      value={{
        collapsed,
        toggleCollapsed: () => setCollapsed(!collapsed)
      }}
    >
      <CardWrapper color={color} {...props}>
        {children}
      </CardWrapper>
    </CardContext.Provider>
  );
};

Card.propTypes = {
  color: PropTypes.oneOf(Object.keys(colors)),
  elevation: PropTypes.number
};

Card.defaultProps = {
  color: 'orange',
  elevation: 1
};

const CardWrapper = styled.div`
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: ${({ elevation, theme, color }) =>
    `${shadows[elevation]}, 0px -8px 0px 0px ${theme.colors[color]}`};
  overflow: hidden;
  transition: box-shadow 0.1s linear;
  margin-top: 8px;
  margin-bottom: ${({ theme }) => `${theme.spacing.unit * 3}px`};

  @media screen and (min-width: 768px) {
    & {
      margin-top: 8px;
    }
  }
`;

/*===========================================
                 CARD HEADER
===========================================*/
const CardHeader = props => {
  const isMobile = !useMediaQuery('(min-width: 768px)');
  const { toggleCollapsed, collapsed } = useContext(CardContext);
  const cardHeaderEl = useRef(null);

  // Auto-expand cards when switching to desktop
  useEffect(() => {
    if (!isMobile && collapsed) {
      toggleCollapsed();
    }
  });

  const handleClick = e => {
    // Remove focus ring on mouse click
    if (e.detail > 0) {
      cardHeaderEl.current.blur();
    }
    toggleCollapsed();
  };

  return isMobile ? (
    <CardHeaderWrapper
      as="button"
      ref={cardHeaderEl}
      className="cardheader-clickable"
      onClick={handleClick}
      {...props}
    />
  ) : (
    <CardHeaderWrapper {...props} />
  );
};

const CardHeaderWrapper = styled.div`
  border: 0;
  background: transparent;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: ${({ theme }) => `0 ${theme.spacing.unit * 2}px`};
  min-height: 68px;
  :hover {
    cursor: ${props => (props.clickable ? 'pointer' : 'initial')};
  }

  &.cardheader-clickable {
    cursor: pointer;
  }
`;

const CardHeaderTitle = styled.div`
  font-size: 24px;
  font-family: 'Stratum2', sans-serif;
  font-weight: bold;
`;

const CardHeaderSubtitle = styled.div`
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  font-weight: 300;
  display: flex;
  align-items: center;
`;

/*===========================================
                 CARD CONTENT
===========================================*/
const CardContentWrapper = styled.div`
  overflow: hidden;
  max-height: ${props => (props.collapsed ? '0' : '100%')};
  padding: ${({ theme, collapsed }) => (collapsed ? 0 : `${theme.spacing.unit * 2}px`)};
  padding-top: ${({ theme, collapsed }) => (collapsed ? 0 : `${theme.spacing.unit}px`)};
`;

const CardContent = props => {
  const { collapsed } = useContext(CardContext);
  return <CardContentWrapper {...props} collapsed={collapsed} aria-expanded={!collapsed} />;
};

export { Card, CardHeader, CardHeaderTitle, CardHeaderSubtitle, CardContent };
