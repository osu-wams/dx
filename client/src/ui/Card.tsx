import React, { useState, useContext, FC } from 'react';
import styled from 'styled-components';
import uuidv4 from 'uuid/v4';
import { faChevronDown, faChevronUp } from '@fortawesome/pro-light-svg-icons';
import Icon from './Icon';
import { Color, shadows, theme } from '../theme';
import useMediaQuery from '../util/useMediaQuery';

const CardContext = React.createContext<any>(null);

const Card = ({ children, ...props }) => {
  // Generate a UUID for linking header to controlled content
  const uuid = uuidv4();
  const [collapsed, setCollapsed] = useState(true);
  const toggleCollapsed = () => setCollapsed(!collapsed);
  const isMobile = !useMediaQuery('(min-width: 768px)');
  const collapsible = isMobile;
  const value = { collapsed, toggleCollapsed, collapsible, uuid };

  return (
    <CardBase style={{ padding: 0 }} {...props}>
      <CardContext.Provider value={value}>{children}</CardContext.Provider>
    </CardBase>
  );
};

const CardBase = styled.div`
  display: flex;
  flex-direction: column;
  border-radius: ${theme.borderRadius};
  box-shadow: ${shadows[1]};
  background-color: ${Color.white};
  overflow: hidden;
  margin-bottom: ${theme.spacing.unit * 2}px;
`;

const CardHeader: FC<{ title: string; badge?: any }> = ({ title, badge, ...props }) => {
  const { collapsed, toggleCollapsed, collapsible, uuid } = useContext(CardContext);
  const handleKeyDown = e => {
    if (e.key === 'Enter' || e.key === ' ') {
      toggleCollapsed();
    }
  };
  return (
    <CardHeaderWrapper
      id={`${uuid}header`}
      collapsible={collapsible}
      collapsed={collapsed}
      onClick={collapsible ? toggleCollapsed : undefined}
      onKeyDown={collapsible ? handleKeyDown : undefined}
      role={collapsible ? 'button' : undefined}
      aria-controls={collapsible ? uuid : undefined}
      aria-expanded={collapsible ? !collapsed : undefined}
      tabIndex={collapsible ? 0 : undefined}
      {...props}
    >
      {badge}
      <span>{title}</span>
      {collapsible && (
        <Icon icon={collapsed ? faChevronDown : faChevronUp} style={{ marginLeft: 'auto' }} />
      )}
    </CardHeaderWrapper>
  );
};

const CardHeaderWrapper = styled.div<{ collapsed: boolean; collapsible: boolean }>`
  height: 64px;
  width: 100%;
  padding: ${theme.spacing.unit * 2}px;
  display: flex;
  align-items: center;
  cursor: ${props => (props.collapsible ? 'pointer' : 'default')};
  border: none;
  border-bottom: ${props =>
    props.collapsed && props.collapsible ? 'none' : `1px solid ${Color['neutral-200']}`};
`;

const CardContent = ({ ...props }) => {
  const { collapsed, collapsible, uuid } = useContext(CardContext);
  return (
    <CardContentWrapper
      id={uuid}
      collapsed={collapsed}
      collapsible={collapsible}
      role={collapsible ? 'region' : undefined}
      aria-labelledby={`${uuid}header`}
      {...props}
    />
  );
};

const CardContentWrapper = styled.div<{ collapsed: boolean; collapsible: boolean }>`
  flex: 1;
  padding: ${theme.spacing.unit * 2}px;
  ${props =>
    props.collapsible &&
    `
    visibility: ${props.collapsed ? 'collapse' : 'visible'};
    height: ${props.collapsed ? 0 : 'auto'};
    padding: ${props.collapsed ? 0 : theme.spacing.unit * 2}px;
  `}
  overflow: hidden;
`;

const CardFooter = ({ ...props }) => {
  const { collapsed, collapsible } = useContext(CardContext);
  return <CardFooterWrapper collapsed={collapsed} collapsible={collapsible} {...props} />;
};

const CardFooterWrapper = styled.div<{ collapsed: boolean; collapsible: boolean }>`
  padding: ${`${theme.spacing.unit}px ${theme.spacing.unit * 2}px`};

  ${props =>
    props.collapsible &&
    `
    display: ${props.collapsed ? 'collapse' : 'visible'}
    height: ${props.collapsed ? 0 : 'auto'};
    padding: ${props.collapsed ? 0 : `${theme.spacing.unit}px ${theme.spacing.unit * 2}px`};
  `}
  overflow: hidden;
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

const Badge = styled.div<{ fg?: Color; bg?: Color }>`
  height: 32px;
  width: 32px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
  background-color: ${props => props.bg || Color['orange-400']};
  color: ${props => props.fg || Color.white};
  margin-right: 12px;
`;

export { Card, CardHeader, CardContent, CardFooter, Badge, CardBase };
