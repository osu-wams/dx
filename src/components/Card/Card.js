import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Paper from '../Paper';

const CardPaper = styled(Paper)`
  transition: box-shadow 0.1s linear;
`;

const Card = ({ raised, elevation, ...rest }) => {
  const cardElevation = raised ? 8 : elevation;
  return <CardPaper elevation={cardElevation} {...rest} />;
};

Card.propTypes = {
  raised: PropTypes.bool
};

Card.defaultProps = {
  raised: false
};

export default Card;
