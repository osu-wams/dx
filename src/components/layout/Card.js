import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import MediaQuery from 'react-responsive';
import { shadows } from '../../theme';
import Badge from './Badge';

const CardWrapper = styled.div`
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: ${({ elevation }) => shadows[elevation]},
    0px -8px 0px 0px ${({ theme, variant }) => theme[variant].bg};
  overflow: hidden;
  transition: box-shadow 0.1s linear;
  margin-top: 8px;
  margin-bottom: ${({ theme }) => `${theme.spacing.unit * 3}px`};

  @media screen and (min-width: 768px) {
    & {
      margin-top: 8px;
      margin-bottom: 0;
    }
  }

  ${Badge} {
    background-color: ${({ variant, theme }) => theme[variant].bg};
    color: ${({ variant, theme }) => theme[variant].fg};
  }

  svg {
    color: ${({ variant, theme }) => theme[variant].bg};
  }
`;

const CardHeader = styled.div`
  display: flex;
  justify-content: space-between;
  padding: ${({ theme }) => `0 ${theme.spacing.unit * 2}px`};
  min-height: 68px;
  line-height: 68px;
  :hover {
    cursor: ${props => (props.clickable ? 'pointer' : 'initial')};
  }

  svg {
    margin-right: ${({ theme }) => `${theme.spacing.unit * 2}px`};
  }
`;

const CardHeaderTitle = styled.div`
  font-size: 24px;
  font-family: 'Stratum2', sans-serif;
  font-weight: bold;
`;

const CardHeaderSubtitle = styled.div`
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  font-weight: 300;
  display: flex;
  align-items: center;
`;

const CardContent = styled.div`
  overflow: hidden;
  max-height: ${props => (props.collapsed ? '0' : '100%')};
  padding: ${({ theme, collapsed }) => (collapsed ? 0 : `${theme.spacing.unit * 2}px`)};
  padding-top: 0;
`;

class Card extends Component {
  state = {
    isCollapsed: true
  };

  collapseContent = isCollapsible => {
    if (isCollapsible) {
      this.setState(state => ({ isCollapsed: !state.isCollapsed }));
    }
  };

  render() {
    const { title, subtitle, headerIcon, variant, children, ...props } = this.props;
    const { isCollapsed } = this.state;

    return (
      <MediaQuery minWidth={768}>
        {isDesktop => (
          <CardWrapper variant={variant} {...props}>
            <CardHeader variant={variant} onClick={this.collapseContent} clickable={!isDesktop}>
              <CardHeaderTitle>
                {headerIcon && <FontAwesomeIcon icon={headerIcon} />}
                {title}
              </CardHeaderTitle>
              <CardHeaderSubtitle>{subtitle}</CardHeaderSubtitle>
            </CardHeader>
            <CardContent collapsed={isCollapsed && !isDesktop}>{children}</CardContent>
          </CardWrapper>
        )}
      </MediaQuery>
    );
  }
}

Card.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.node,
  variant: PropTypes.oneOf(['primary', 'secondary', 'academic']),
  elevation: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7, 8])
};

Card.defaultProps = {
  subtitle: '',
  variant: 'primary',
  elevation: 1
};

export default Card;
