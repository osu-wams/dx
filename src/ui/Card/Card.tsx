import React, { useState, FC } from 'react';
import uuidv4 from 'uuid/v4';
import { useMediaQuery } from 'react-responsive';
import { CardBase } from './StyledCardComponents';
import { breakpoints } from 'src/theme';

const CardContext = React.createContext<any>(null);

const Card: FC<{ collapsing?: boolean }> = ({ children, collapsing = true, ...props }) => {
  // Generate a UUID for linking header to controlled content
  const uuid = uuidv4();
  const [collapsed, setCollapsed] = useState(true);
  const toggleCollapsed = () => setCollapsed(!collapsed);
  const isMobile = !useMediaQuery({ minWidth: breakpoints.small });
  const collapsible = isMobile && collapsing;
  const value = { collapsed, toggleCollapsed, collapsible, uuid };

  return (
    <CardBase style={{ padding: 0 }} {...props}>
      <CardContext.Provider value={value}>{children}</CardContext.Provider>
    </CardBase>
  );
};

export { Card, CardContext };
