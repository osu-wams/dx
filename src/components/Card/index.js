import Card from './Card';
import CardHeader from './CardHeader';
import CardContent from './CardContent';

export { CardHeader, CardContent };

export default Card;
