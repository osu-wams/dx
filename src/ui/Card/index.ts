import { Card, CardContext } from './Card';
import CardIcon from './CardIcon';
import CardFooter from './CardFooter';
import CardContent from './CardContent';
import CardHeader from './CardHeader';
import {
  CardBase,
  CardButtonBase,
  Badge,
  CardContentCell,
  CardContentRow,
  CardContentTable,
} from './StyledCardComponents';

export {
  Card,
  CardHeader,
  CardContent,
  CardFooter,
  Badge,
  CardIcon,
  CardBase,
  CardContentTable,
  CardContentRow,
  CardContentCell,
  CardContext,
  CardButtonBase,
};
