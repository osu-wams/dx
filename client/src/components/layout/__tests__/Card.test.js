import { render, fireEvent, wait } from '../../../componentTestUtils';
import React from 'react';
import { Card, CardHeader, CardHeaderTitle, CardHeaderSubtitle, CardContent } from '../Card';

const DefaultCard = () => (
  <Card>
    <CardHeader>
      <div>
        <CardHeaderTitle>Default Card</CardHeaderTitle>
        <CardHeaderSubtitle>Subtitle</CardHeaderSubtitle>
      </div>
    </CardHeader>
    <CardContent>
      <p>DefaultCard Content</p>
    </CardContent>
  </Card>
);

const StratosphereCard = () => (
  <Card color="stratosphere">
    <CardHeader>
      <div>
        <CardHeaderTitle>Stratosphere Card</CardHeaderTitle>
        <CardHeaderSubtitle>Subtitle</CardHeaderSubtitle>
      </div>
    </CardHeader>
    <CardContent>
      <p>DefaultCard Content</p>
    </CardContent>
  </Card>
);

const SimpleCard = () => (
  <Card>
    <CardHeader>
      <div>
        <CardHeaderTitle>Header Button</CardHeaderTitle>
        <CardHeaderSubtitle>Subtitle</CardHeaderSubtitle>
      </div>
    </CardHeader>
    <CardContent>
      <p>Card Content</p>
    </CardContent>
  </Card>
);

it('DefaultCard card styles and structure are present', () => {
  const { container } = render(<DefaultCard />);
  expect(container.firstChild).toMatchSnapshot();
});

it('Stratosphere card styles and structure are present', () => {
  const { container } = render(<StratosphereCard />);
  expect(container.firstChild).toMatchSnapshot();
});

it('SimpleCard Orange card styles and structure are present', () => {
  const { container } = render(<SimpleCard />);
  expect(container.firstChild).toMatchSnapshot();
});

it('aria-expanded toggles on click', async () => {
  const { getByText } = render(<SimpleCard />);

  // Confirm the cardheader-clickable class is present on the button
  const button = getByText(/header button/i).parentElement.parentElement;
  await wait(() => expect(button).toHaveClass('cardheader-clickable'));
  // Card content is hidden by default
  const cardContent = getByText('Card Content').parentElement;
  expect(cardContent).toHaveAttribute('aria-expanded', 'false');

  // Clicking the button sets aria-expanded to true
  fireEvent.click(button);
  expect(cardContent).toHaveAttribute('aria-expanded', 'true');

  // Clicking again sets it to false
  fireEvent.click(button);
  expect(cardContent).toHaveAttribute('aria-expanded', 'false');
});
