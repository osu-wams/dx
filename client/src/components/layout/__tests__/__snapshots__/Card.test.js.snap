// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultCard card styles and structure are present 1`] = `
.c0 {
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 2px 1px -1px rgba(0,0,0,0.12),0px -8px 0px 0px #d73f09;
  overflow: hidden;
  -webkit-transition: box-shadow 0.1s linear;
  transition: box-shadow 0.1s linear;
  margin-top: 8px;
  margin-bottom: 24px;
}

.c1 {
  border: 0;
  background: transparent;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  min-height: 68px;
}

.c1:hover {
  cursor: initial;
}

.c1.cardheader-clickable {
  cursor: pointer;
}

.c2 {
  font-size: 24px;
  font-family: 'Stratum2',sans-serif;
  font-weight: bold;
}

.c3 {
  font-family: 'Open Sans',sans-serif;
  font-size: 16px;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  overflow: hidden;
  max-height: 0;
  padding: 0;
  padding-top: 0;
}

@media screen and (min-width:768px) {
  .c0 {
    margin-top: 8px;
  }
}

<div
  class="c0"
  color="orange"
  elevation="1"
>
  <button
    class="cardheader-clickable c1"
  >
    <div>
      <div
        class="c2"
      >
        Default Card
      </div>
      <div
        class="c3"
      >
        Subtitle
      </div>
    </div>
  </button>
  <div
    aria-expanded="false"
    class="c4"
  >
    <p>
      DefaultCard Content
    </p>
  </div>
</div>
`;

exports[`SimpleCard Orange card styles and structure are present 1`] = `
.c0 {
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 2px 1px -1px rgba(0,0,0,0.12),0px -8px 0px 0px #d73f09;
  overflow: hidden;
  -webkit-transition: box-shadow 0.1s linear;
  transition: box-shadow 0.1s linear;
  margin-top: 8px;
  margin-bottom: 24px;
}

.c1 {
  border: 0;
  background: transparent;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  min-height: 68px;
}

.c1:hover {
  cursor: initial;
}

.c1.cardheader-clickable {
  cursor: pointer;
}

.c2 {
  font-size: 24px;
  font-family: 'Stratum2',sans-serif;
  font-weight: bold;
}

.c3 {
  font-family: 'Open Sans',sans-serif;
  font-size: 16px;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  overflow: hidden;
  max-height: 0;
  padding: 0;
  padding-top: 0;
}

@media screen and (min-width:768px) {
  .c0 {
    margin-top: 8px;
  }
}

<div
  class="c0"
  color="orange"
  elevation="1"
>
  <button
    class="cardheader-clickable c1"
  >
    <div>
      <div
        class="c2"
      >
        Header Button
      </div>
      <div
        class="c3"
      >
        Subtitle
      </div>
    </div>
  </button>
  <div
    aria-expanded="false"
    class="c4"
  >
    <p>
      Card Content
    </p>
  </div>
</div>
`;

exports[`Stratosphere card styles and structure are present 1`] = `
.c0 {
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 2px 1px -1px rgba(0,0,0,0.12),0px -8px 0px 0px #006a8e;
  overflow: hidden;
  -webkit-transition: box-shadow 0.1s linear;
  transition: box-shadow 0.1s linear;
  margin-top: 8px;
  margin-bottom: 24px;
}

.c1 {
  border: 0;
  background: transparent;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  min-height: 68px;
}

.c1:hover {
  cursor: initial;
}

.c1.cardheader-clickable {
  cursor: pointer;
}

.c2 {
  font-size: 24px;
  font-family: 'Stratum2',sans-serif;
  font-weight: bold;
}

.c3 {
  font-family: 'Open Sans',sans-serif;
  font-size: 16px;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  overflow: hidden;
  max-height: 0;
  padding: 0;
  padding-top: 0;
}

@media screen and (min-width:768px) {
  .c0 {
    margin-top: 8px;
  }
}

<div
  class="c0"
  color="stratosphere"
  elevation="1"
>
  <button
    class="cardheader-clickable c1"
  >
    <div>
      <div
        class="c2"
      >
        Stratosphere Card
      </div>
      <div
        class="c3"
      >
        Subtitle
      </div>
    </div>
  </button>
  <div
    aria-expanded="false"
    class="c4"
  >
    <p>
      DefaultCard Content
    </p>
  </div>
</div>
`;
